pynxtools/dataconverter/helpers.py

```diff
@@ -650,3 +651,72 @@ def extract_atom_types(formula, mode='hill'):
         return convert_to_hill(atom_types)
 
     return atom_types
+
+
+def transform_to_intended_dt(str_value: Any) -> Optional[Any]:
```

Comment:
Can you also write tests for this?

Also I think this can be integrated with the is_valid_data_field function.